app/gui/view/component-browser/searcher-list-panel/src/lib.rs

```diff
@@ -0,0 +1,509 @@
+        section
+    }
+
+    fn init_wide_section(app: &Application, label: &str) -> WideSection {
+        let content = app.new_view::<component_group::wide::View>();
+        content.set_no_items_label_text("No Entries.");
+        content.set_position_x(PADDING_INNER + CONTENT_WIDTH / 2.0);
+        content.set_width(WIDTH_INNER - 2.0 * PADDING_INNER);
+        let section = LabeledSection::new(content, app);
+        section.label.set_content(label);
+        section
+    }
+
+    fn update_layers(&self) {
+        self.favourites_section.set_layers(&self.layers, self.scroll_area.content_layer());
+        self.local_scope_section.set_layers(&self.layers, self.scroll_area.content_layer());
+        self.sub_modules_section.set_layers(&self.layers, self.scroll_area.content_layer());
+    }
+
+    fn recompute_layout(&self) {
+        self.update_layers();
+
+        let favourites_section_height = self.favourites_section.height();
+        let local_scope_height = self.local_scope_section.height();
+        let sub_modules_height = self.sub_modules_section.height();
+
+        self.favourites_section.set_base_position_y(0.0);
+        self.local_scope_section.set_base_position_y(-favourites_section_height);
+        self.sub_modules_section
+            .set_base_position_y(-favourites_section_height - local_scope_height);
+
+        self.scroll_area.set_content_height(
+            favourites_section_height + local_scope_height + sub_modules_height,
+        );
+    }
+}
+
+impl display::Object for Model {
+    fn display_object(&self) -> &display::object::Instance {
+        &self.display_object
+    }
+}
+
+impl component::Model for Model {
+    fn label() -> &'static str {
+        "ComponentBrowserPanel"
+    }
+
+    fn new(app: &Application, _logger: &DefaultWarningLogger) -> Self {
+        Self::new(app)
+    }
+}
+
+
+
+// =======================
+// === Labeled Section ===
+// =======================
+
+/// Struct that contains the components contained in a section of the Component Browser Panel. Also
+/// provides some utility functions for shape and layout handling.
+#[derive(Clone, Debug, CloneRef)]
+struct LabeledSection<T: CloneRef> {
+    pub label:   text::Area,
+    pub divider: hline::View,
+    pub content: T,
+}
+
+type WideSection = LabeledSection<component_group::wide::View>;
+type ColumnSection = LabeledSection<column_grid::ColumnGrid>;
+
+impl<T: CloneRef> LabeledSection<T> {
+    pub fn new(content: T, app: &Application) -> Self {
+        let logger = Logger::new("LabeledSection");
+        let label = text::Area::new(app);
+        let divider = hline::View::new(logger);
+        divider.size.set(Vector2(INFINITE, SECTION_DIVIDER_HEIGHT));
+
+        Self { label, divider, content }.init_label()
+    }
+
+    fn init_label(self) -> Self {
+        let label = &self.label;
+        let section_header_color: color::Rgba =
+            color::Rgb::from_css_hex(SECTION_HEADING_COLOR_HEX).unwrap().into();
```

Comment:
Don't use `unwrap` in the code. If user defines a theme and provides wrng color name, you dont want the app to crash ...

Reply:
Note that in this case, this is unwrapping a constant. With a theme, you are completely right (and this will be changed while refactoring to a theme.